lib/gossip/dissemination.js

```diff
@@ -224,7 +216,74 @@ Dissemination.prototype._issueAs = function _issueAs(filterChange, mapChanges) {
         });
     }
 
-    return mapChanges(changesToDisseminate);
+    return changesToDisseminate;
+};
+
+Dissemination.prototype.tryStartReverseFullSync = function tryStartFullSync(target, timeout) {
+    var self = this;
+
+    if (this.reverseFullSyncJobs >= this.ringpop.maxReverseFullSyncJobs) {
+        this.logger.info('omit bidirectional full sync, too many already running', {
+            'remote': target,
+            'reverseFullSyncJobs': this.reverseFullSyncJobs,
+            'maxReverseFullSyncJobs': this.ringpop.maxReverseFullSyncJobs
+        });
+        this.ringpop.stat('increment', 'full-sync.reverse-omitted');
+        return;
+    }
+
+    this.reverseFullSyncJobs += 1;
+    this.ringpop.stat('gauge', 'full-sync.reverse-jobs', this.reverseFullSyncJobs);
+
+    var start = new Date();
+    this._reverseFullSync(target, timeout, onReverseFullSync);
+
+    function onReverseFullSync() {
+        self.ringpop.stat('timing', 'full-sync.reverse', start);
+        self.reverseFullSyncJobs -= 1;
```

Comment:
I like how you use a callback for decrementing and keep this fields use limited to this function